heading_controller:
  ros__parameters:
    sailbot:
      heading_kp: 0.5
    
esp32_comms:
  ros__parameters:
    sailbot:
      rudder:
        angle_limit_deg: 30

path_follower:
  ros__parameters:
    sailbot:
      pathfinding:
        buoy_rounding_distance_meters: 5.0
        buoy_threat_size_map_units: 2.0
        buoy_threat_guassian_intensity: 0.6
        min_path_recalculation_interval_seconds: 10.0
      navigation:
        look_ahead_distance_meters: 3.0
        look_ahead_increase_per_knot: 1.0
        buoy_snap_distance_meters: 10.0

wind_smoother:
  ros__parameters:
    sailbot:
      num_wind_readings: 20

buoy_detection:
  ros__parameters:
    sailbot:
      cv:
        lower_h: 3
        lower_s: 180
        lower_v: 74
        upper_h: 15
        upper_s: 255
        upper_v: 255
        buoy_circularity_threshold: 0.4
        buoy_diameter_meters: 0.5